connection_manager.go

```diff
@@ -217,8 +269,10 @@ func (cm *connectionManager) get() (*connection, error) {
 			// we found our connection, don't re-queue
 			return true, false
 		} else {
-			// keep going and re-queue conn
-			return false, true
+			// Remove connection, don't re-queue, keep going
+			cm.remove(conn)
+			conn = nil // GH-47
+			return false, false
```

Comment:
The changes here were the main fix to the bug. Previously, while iterating through the pool to find an available connection, if the connection were marked `connInactive` it was re-enqueued. Here, I remove the connection from the pool as well as ensure the captured `conn` variable is nullified just in case this was the last connection to be checked. Without setting `conn` to `nil`, the routine would continue past the iterator and think that an available connection had been found. This would prevent new connections from being created as needed, and using this connection would cause a TCP error.